src/chrome/chromeDebugAdapter.ts

```diff
@@ -1140,10 +1162,13 @@ export abstract class ChromeDebugAdapter implements IDebugAdapter {
                     });
 
                     // Return the setBP request, no matter how long it takes. It may take awhile in Node 7.5 - 7.7, see https://github.com/nodejs/node/issues/11589
-                    return setBreakpointsPFailOnError.then(body => {
+                    return setBreakpointsPFailOnError.then(setBpResultBody => {
+                        const body = { breakpoints: setBpResultBody.breakpoints.map(setBpResult => setBpResult.breakpoint) };
                         if (body.breakpoints.every(bp => !bp.verified)) {
+                            // If all breakpoints are set, we mark them as set. If not, we mark them as un-set so they'll be set
+                            const areAllSet = setBpResultBody.breakpoints.every(setBpResult => setBpResult.isSet);
```

Comment:
I think this changes the way it works - bpsSet was true if the bp was attempted to be set - not that it was bound. The check below `response.breakpointId !== undefined` is only true when the bp is bound.

Reply:
My understanding was that Debugger.setBreakpointByUrl always returns the breakpointId if the parameters are valid even if the breakpoint is unbound, and it also returns the actualLocation if the breakpoint was bound.

Reply:
Sorry, you're correct.